code/game/objects/effects/decals/Cleanable/fuel.dm

```diff
@@ -90,6 +96,26 @@
 			INVOKE_NEXT_TICK(other, .proc/ignite_fuel)	//Spread effect
 	qdel(src)
 
+///Ignites when something hot enters our loc, either via crossed or signal wrapper
+/obj/effect/decal/cleanable/liquid_fuel/proc/check_ignite(atom/movable/igniter)
+	if(isitem(igniter))
+		var/obj/item/ignitem = igniter
+		if(ignitem.damtype != BURN)
+			return
+	else if(isliving(igniter))
+		var/mob/living/burner_mob = igniter
+		if(!burner_mob.on_fire)
+			return
+	else
+		return
+	visible_message(span_warning("The spilled fuel catches fire!"))
```

Comment:
this shouldnt be outside the ignite fuel proc because there are multiple wys to ignite
```suggestion
```

Reply:
Ignite_fuel calls ignite_fuel on all adjacent puddles, if the visible_message is there it'll get spammed. The three ignite options are this, attack_by which has its own message, and adjacent ignites.

Reply:
Yes which is why it shouldnt be this inconsistent, either it shuld always work or not at all